dmutils/timing.py

```diff
@@ -118,3 +99,58 @@ def logged_duration(
 logged_duration.default_condition = _logged_duration_default_condition
 # exposing this allows a caller to specify default_logger.getChild(...) as their logger
 logged_duration.default_logger = _logged_duration_default_logger
+
+
+def exceeds_slow_external_call_threshold(log_context):
+    """A public condition that will return True if the duration is above the threshold we have defined as acceptable for
+    calls to external services (e.g. Notify, Mailchimp, S3, etc)."""
+    return log_context['duration_real'] > SLOW_EXTERNAL_CALL_THRESHOLD
+
+
+def request_is_sampled(log_context):
+    """A public condition that returns True if the request has the X-B3-Sampled flag set in its headers. While this is
+    the default condition for logged_duration, exposing it publically allows it to be easily combined with other
+    conditions."""
+    return has_request_context() and getattr(request, "is_sampled", False)
+
+
+def different_message_for_success_or_error(success_message, error_message):
+    """Can be passed into `logged_duration` as `message=different_message_for_success_or_error(x, y)` in order to
+    generate different log messages depending on whether the block completed successfully or raised an exception."""
+    return lambda _: success_message if sys.exc_info()[0] is None else error_message
+
+
+def partial_logged_duration_for_external_request(service, description=None, success_message=None, error_message=None):
```

Comment:
Might not need the `partial_` in the name here - will probably confuse people and to some extent it's an "implementation detail"

Reply:
I included it because I don't think it's immediately apparent that you need to call `()` the result of this function otherwise. I suspect it would be more confusing with calling out that it's a partial, but I'm open to being convinced otherwise.

Reply:
Oh no I see what you mean by it now. Do you think it would be simpler to just merge the two argument lists and avoid the (potentially confusing?) double-function-call by actually calling `logged_duration` yourself? Don't know.

Reply:
Good idea, I've removed the partial-ness.